eos-tech-support/eos-core-dump

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -e
+#
+# This program is used to enable core dumps on the system
+
+CORE_DIR="/var/coredumps"
+KERNEL_CORE_PATTERN='core.%t.%E.%p'
+
+# TODO: Implement switches for permanent changes
+if [ $# != 0 ]; then
+  echo "Usage: $0"
+  exit 1
+fi
+
+if [ ! $EUID == 0 ]; then
+  echo "Must be run with sudo privileges! Exiting!"
+  exit 1
+fi
+
+echo "*** Changes do not persist across reboots!! ***"
+echo "*** Changes do not affect daemons!! ***"
+echo
+
+echo "Creating coredump dir in $CORE_DIR (if needed)..."
+mkdir -p $CORE_DIR
+
+echo "Setting/fixing directory permissions..."
+chown root:root "${CORE_DIR}"
+chmod 1777 "${CORE_DIR}"
+
+echo "Changing kernel core settings..."
+sysctl kernel.core_pattern="${CORE_DIR}/${KERNEL_CORE_PATTERN}" 1> /dev/null
+sysctl kernel.core_pipe_limit=4 > /dev/null
+sysctl kernel.core_uses_pid=1 > /dev/null
+sysctl fs.suid_dumpable=1 > /dev/null
+
+echo
+echo "Changing all process ulimits..."
```

Comment:
Hmm, this only affects current processes and won't change ulimits for new processes. I suppose since `pam_limits` will only be invoked on certain processes, they'll just be inherited from their parent processes. I wonder if we even need to do this, though. Looking at `/etc/security/limits.conf`, there are no settings. What does `ulimit -c` report without any changes?